### PR TITLE
MNT: Fix test compat with scipy 1.10.0

### LINKED ISSUE #13644: TST: scipy.misc.ascent has been deprecated in SciPy v1.10.0
Example log: https://github.com/astropy/astropy/runs/8299487818?check_suite_focus=true

```
______________ TestCompressedImage.test_comp_image_quantize_level ______________

self = <astropy.io.fits.tests.test_image.TestCompressedImage object at 0x7fd57406a650>

    @pytest.mark.skipif('not HAS_SCIPY')
    def test_comp_image_quantize_level(self):
        """
        Regression test for https://github.com/astropy/astropy/issues/5969
    
        Test that quantize_level is used.
    
        """
        import scipy.misc
        np.random.seed(42)
>       data = scipy.misc.ascent() + np.random.randn(512, 512)*10

.../astropy/io/fits/tests/test_image.py:1137: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

args = (), kwargs = {}

    @functools.wraps(fun)
    def call(*args, **kwargs):
>       warnings.warn(msg, category=DeprecationWarning,
                      stacklevel=stacklevel)
E       DeprecationWarning: scipy.misc.ascent has been deprecated in SciPy v1.10.0;
and will be completely removed in SciPy v1.12.0. Dataset methods have moved
into the scipy.datasets module. Use scipy.datasets.ascent instead.

.../scipy/_lib/deprecation.py:18: DeprecationWarning
```

### PR DESCRIPTION
**Pull request by pllim**
### Description

This pull request is to address compatibility issue with scipy dev.

Fix #13644

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Do the proposed changes need the code-style [fixed](https://docs.astropy.org/en/latest/development/workflow/maintainer_workflow.html#pre-commit_bot)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [x] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [x] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

**Comment by pllim** [comment]
`ImportError: Missing optional dependency 'pooch' required for scipy.datasets module. Please use pip or conda to install 'pooch'.` -- Wut?

File: astropy/io/fits/tests/test_image.py [hunk 1]
```diff
@@ -1132,17 +1132,26 @@ def test_comp_image_quantize_level(self):
...
+
+        SCIPY_LT_1_10 = not minversion(scipy, '1.10dev0')
         np.random.seed(42)
-        data = scipy.misc.ascent() + np.random.randn(512, 512)*10
+
+        if SCIPY_LT_1_10:
+            import scipy.misc
+            scipy_data = scipy.misc.ascent()
+        else:
+            import scipy.datasets
```
Comment (Cadair) [inline astropy/io/fits/tests/test_image.py:1145]: Does this download data?

Comment (pllim) [reply astropy/io/fits/tests/test_image.py:1145]: Yes, I think it downloads the "ascent" image data from their repo.

Comment (pllim) [reply astropy/io/fits/tests/test_image.py:1145]: I think? https://github.com/scipy/scipy/blob/06da9ee65cdee4a8aa036767489bfc5a030527f4/scipy/datasets/_fetchers.py#L76

Hmm, should we mark this as remote data?

Comment (Cadair) [reply astropy/io/fits/tests/test_image.py:1145]: > Hmm, should we mark this as remote data?

This was my thinking, but then I am surprised it worked without that.

Comment (pllim) [reply astropy/io/fits/tests/test_image.py:1145]: https://raw.githubusercontent.com/scipy/dataset-ascent/main/ascent.dat

Comment (pllim) [reply astropy/io/fits/tests/test_image.py:1146]: @Cadair , looks better?

**Pull request merged**